sklearn/utils/validation.py

```diff
@@ -374,6 +378,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         if force_all_finite:
             _assert_all_finite(array)
 
+    if array.ndim == 1:
```

Comment:
If I remove the `np.atleast_2d` call I get 227 errors in the tests. It's my understanding that this PR should fix all of them ?